python/ray/serve/_private/common.py

```diff
@@ -102,30 +105,53 @@ class DeploymentStatus(str, Enum):
     UPDATING = "UPDATING"
     HEALTHY = "HEALTHY"
     UNHEALTHY = "UNHEALTHY"
+    UPSCALING = "UPSCALING"
+    DOWNSCALING = "DOWNSCALING"
+
+
+class DeploymentStatusTrigger(str, Enum):
+    UNSPECIFIED = "UNSPECIFIED"
+    DEPLOY = "DEPLOY"
+    CONFIG_UPDATE = "CONFIG_UPDATE"
+    UPSCALE = "UPSCALE"
+    DOWNSCALE = "DOWNSCALE"
+    AUTOSCALE = "AUTOSCALE"
+    DELETE = "DELETE"
 
 
 @dataclass(eq=True)
 class DeploymentStatusInfo:
     name: str
     status: DeploymentStatus
+    status_trigger: DeploymentStatusTrigger = DeploymentStatusTrigger.UNSPECIFIED
     message: str = ""
 
     def debug_string(self):
         return json.dumps(asdict(self), indent=4)
 
+    def update(self, **kwargs):
+        for key, value in kwargs.items():
+            if hasattr(self, key):
+                setattr(self, key, value)
```

Comment:
can't you just set the fields directly? e.g., `status.name = "whatever"`

Reply:
Switched to update because previously we were reconstructing a new `DeploymentStatusInfo` each time we modify the status but sometimes I want to just update one field. I made a method for it because sometimes I want to update multiple fields at the same time so it's cleaner imo to have an update method.